pkg/trace/stats/concentrator.go

```diff
@@ -113,28 +116,46 @@ type Input struct {
 	Traces        []traceutil.ProcessedTrace
 	ContainerID   string
 	ContainerTags []string
+	ProcessTags   string
 }
 
 // NewStatsInput allocates a stats input for an incoming trace payload
-func NewStatsInput(numChunks int, containerID string, clientComputedStats bool) Input {
+func NewStatsInput(numChunks int, containerID string, clientComputedStats bool, processTags string) Input {
 	if clientComputedStats {
 		return Input{}
 	}
-	return Input{Traces: make([]traceutil.ProcessedTrace, 0, numChunks), ContainerID: containerID}
+	return Input{Traces: make([]traceutil.ProcessedTrace, 0, numChunks), ContainerID: containerID, ProcessTags: processTags}
 }
 
 // Add applies the given input to the concentrator.
 func (c *Concentrator) Add(t Input) {
+	tags := infraTags{
+		containerID:     t.ContainerID,
+		containerTags:   t.ContainerTags,
+		processTagsHash: processTagsHash(t.ProcessTags),
+		processTags:     t.ProcessTags,
+	}
 	for _, trace := range t.Traces {
-		c.addNow(&trace, t.ContainerID, t.ContainerTags)
+		c.addNow(&trace, tags)
 	}
 }
 
+type infraTags struct {
+	containerID     string
+	containerTags   []string
+	processTagsHash uint64
+	processTags     string
```

Comment:
nit: a bit stange to have `containerTags` being a slice and `processTags` a string, is it because you don't want to perform useless conversions? I assume converting `containerTags` to a string (since they will end up concatenated anyway) would cause too many changes?

Reply:
given that other products use the trace-agent as a proxy, I wanted to keep the same encoding like this we can do same normalization/treatments in different ingestion points